### PR TITLE
Move testing to a separate utility

### PR DESCRIPTION
**Pull request by nafur**
This PR moves the management of temporary files and everything related to running the actual checks to separate utilities.
Also, it actually sets up the logging...

### CODE REVIEW

File: ddsmt.py [hunk 1]
```diff
@@ -59,7 +61,7 @@ def setup_logging():
         1: logging.INFO,
         2: logging.DEBUG,
     }
-    logging.getLogger().setLevel(level=verbositymap.get(g_args.verbosity, logging.DEBUG))
+    logging.getLogger().setLevel(level=verbositymap.get(options.args().verbosity, logging.DEBUG))
```
Comment (mpreiner) [inline ddsmt.py:64]: 80 chars

File: utils/tmpfiles.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    options.args().cmd[0] = __BINARY
+
+    if options.args().cmd_cc:
+        shutil.copy(options.args().cmd_cc[0], __BINARY_CC)
+        options.args().cmd_cc[0] = __BINARY_CC
+
+
+def get_tmp_filename():
+    """Return a filename within our temporary directory based on the pid."""
+    return os.path.join(__TMPDIR.name, "ddsmt-tmp-{}.smt2".format(os.getpid()))
```
Comment (mpreiner) [inline utils/tmpfiles.py:27]: 80 chars

**Review by mpreiner — APPROVED** [review]
LGTM. Just make sure to have 80 chars in a line.

File: ddsmt.py (hunk 1 above)
Comment (nafur) [reply ddsmt.py]: Fixed.

**Pull request merged**